pkg/labels/consul_applicator.go

```diff
@@ -201,9 +201,23 @@ func (c *consulApplicator) mutateLabels(labelType Type, id string, labels map[st
 	return nil
 }
 
-// TODO: replace mutateLabels() with this transaction-using implementation
-func (c *consulApplicator) mutateLabelsTxn(ctx context.Context, labelType Type, id string, labels map[string]*string) error {
-	l, index, err := c.GetLabelsWithIndex(labelType, id)
+type LabelFetcher interface {
+	GetLabelsWithIndex(labelType Type, id string) (Labeled, uint64, error)
+}
+
+// mutateLabelsTxn adds operations to the transaction within the passed context to safely
+// make the label mutations requested. It's written as a package level function to avoid
```

Comment:
oh RemoveLabelTxn shouldn't say it's package level I originally was going to write it that way so that you have to pass an applicator as an argument but decided it's more cumbersome that way. maybe i'll say "package global" instead of "package level" as well.